blockchain/difficulty.go

```diff
@@ -540,6 +540,34 @@ func (b *BlockChain) sumPurchasedTickets(startNode *blockNode, numToSum int64) i
 	return numPurchased
 }
 
+// LiveTicketsAtNextMaturity returns the total number of live tickets
+// available after the ticket maturity period from the provided block.
+// This assumes all blocks mined during the period are voted on
+// by the maximum number of tickets allowed per block.
+func (b *BlockChain) LiveTicketsAtNextMaturity(hash *chainhash.Hash) uint64 {
+	b.chainLock.Lock()
+	defer b.chainLock.Unlock()
+
+	node := b.index.lookupNode(hash)
+	liveTickets := int64(node.stakeNode.PoolSize())
+	maturityWindow := int64(b.chainParams.TicketMaturity) + 1
+	rangeStart := node.height - maturityWindow
+
+	// Iterate through all nodes in the last ticket maturity period to
+	// get the number of tickets maturing in the next ticket maturity period.
+	for ; node != nil && node.height >= rangeStart; node = node.parent {
+		liveTickets += int64(node.freshStake)
+	}
+
+	// Subtract the total number of live tickets needed to progress the
+	// chain to the end of the ticket maturity period from the current tip.
+	liveTickets -= maturityWindow * int64(b.chainParams.TicketsPerBlock)
```

Comment:
This logic is incorrect prior to SVH.  There are no votes being consumed.  It's important to count this correctly, otherwise the check won't work properly pre SVH.